### DISCUSSION TITLE
Keep the original text color in a deactivated / disabled button

### DISCUSSION THREAD

**Discussion by ruanformigoni** [fltk-rs/fltk-rs#1310] [Q&A] [answered]
Hi, is there a way to keep the original text color in a deactivated / disabled button?

I tried this:
```rust
    let mut output = Output::new(self.border, 0, self.width - self.border*2, 100, "");
    output.deactivate();
    output.set_text_size(20);
    output.set_value("App");
    output.set_frame(FrameType::NoBox);
    output.set_text_color(Color::White);
```

But the text is still grayed out.

Thanks,

**Comment by MoAlyousef** [comment, marked as answer]
Hi

You would probably have to do so in the button’s draw method. I don’t have a pc to try things out but I’ll post some code when I do. 

For outputs as in your example, why do you want to deactivate it?
You could probably acheive your aim by just changing the output’s color and text_color and use a box type for the frame (FlatBox for example).

**Reply by ruanformigoni** [reply]
Thanks for answering, I want to make it un-selectable.

Enabled:
![image](https://user-images.githubusercontent.com/18180693/201064521-f053666a-076c-4c8b-a83d-5b0a0d646253.png)

Disabled:
![image](https://user-images.githubusercontent.com/18180693/201064694-1d550319-d406-42fd-beea-23f7d2e87d65.png)

**Reply by MoAlyousef** [reply]
You might prefer to use a Frame instead of an Output:
```rust
    let mut output = Frame::new(self.border, 0, self.width - self.border*2, 100, "App");
    output.set_label_size(20);
    output.set_frame(FrameType::NoBox);
    output.set_label_color(Color::White);
    output.set_align(Align::Left | Align::Inside);
```

For a button, overriding the draw method seems to work:
```rust
    let mut button = Button::new(100, 100, 160, 80, "Button");
    button.set_label_color(Color::White);
    button.deactivate();
    button.draw(|b| {
        draw::set_font(Font::Helvetica, 14);
        draw::set_draw_color(Color::White);
        draw::draw_text2(&b.label(), b.x(), b.y(), b.w(), b.h(), Align::Center);
    });
```

**Reply by ruanformigoni** [reply]
It worked perfectly, thanks!
